src/app/(admin)/admin/blog/all/page.tsx

```diff
@@ -1,5 +1,99 @@
-const page = () => {
-  return <div>Hello World</div>;
+"use client";
+import { Container, Typography } from "@mui/material";
+import { GridColDef } from "@mui/x-data-grid";
+import { deleteBlogDB, deleteBlogStorage, getBlogs } from "api/blog";
+import EditModal from "components/blog/EditModal";
+import { AdminDataGrid, EditDeleteActions } from "components/data-grid-custom";
+import {
+  ConfirmationDialog,
+  LoadingScreen,
+} from "components/global-components";
+import { useSession } from "next-auth/react";
+import { enqueueSnackbar } from "notistack";
+import { useEffect, useState } from "react";
+import { IBlogPage } from "types/blog";
+
+const BlogEditTable = () => {
+  const [blogToDelete, setBlogToDelete] = useState<IBlogPage | null>(null);
+  const [blogs, setBlogs] = useState<IBlogPage[]>([]);
+  const [editBlog, setEditBlog] = useState<IBlogPage | null>(null);
+  const { data: session } = useSession();
+
+  useEffect(() => {
+    // add event listener on firestore collectionxX
+    const unsubscribe = getBlogs(setBlogs);
+    // remove event listener on unmount
+    return () => unsubscribe();
+  }, []);
+
+  const columns: GridColDef[] = [
+    {
+      field: "title",
+      headerName: "Title",
+      width: 500,
+    },
+    { field: "author", headerName: "Author", width: 120 },
+    {
+      field: "created",
+      headerName: "Created",
+      width: 200,
+      renderCell: (params) => new Date(params.value).toLocaleString(),
+    },
+
+    {
+      field: "actions",
+      headerName: "Actions",
+      width: 100,
+      renderCell: (params) => (
+        <EditDeleteActions
+          session={session}
+          onDeleteClick={() => setBlogToDelete(params.row)}
+          onEditClick={() => {
+            setEditBlog(params.row);
+          }}
+        />
+      ),
+    },
+  ];
+
+  const handleDeleteblog = async () => {
+    if (!blogToDelete) return;
+
+    deleteBlogStorage(blogToDelete.imageId, blogToDelete.slug).catch(() =>
+      enqueueSnackbar("Error deleting hero image")
+    );
+    deleteBlogDB(blogToDelete.slug)
+      .then(() => {
+        enqueueSnackbar("Blog post deleted");
+      })
+      .catch(() => enqueueSnackbar("Error deleting test"))
+      .finally(() => {
+        setBlogToDelete(null);
+      });
+  };
+
+  if (!session) return <LoadingScreen />;
+
+  if (!blogs) return <LoadingScreen />;
```

Comment:
Combine these two

`if (!session && !blogs) {}`

not sure if its & or && but see if typescript gets angry